### ISSUE TITLE
Possible bug when <style> tag is at the root

### ISSUE THREAD

**Issue by patrick-steele-idem** [marko-js/marko#690]
Given the following template:

```marko
<style> 
#search-guide { 


} 
</style> 
<div id="search-guide"> 
  $ console.log( 'search-guide::state - ', state ); 
  <div> 
    <!-- state.len: ${ state.len } --> 
  </div> 

    <button on-click('pickSomething')>Pick Something!</button> 

    <hr> 

    Picked: ${ state.len } 
    Pick Something Else! 
    <!-- <button on-click('pickSomething')>Pick Something!</button> --> 
</div>
```

## From @ramses0:

...this gets stuck in an infinite loop after the on-click, due to it getting stuck in the weeds with the `style` tag during    `$__rerender: function(isRerenderInBrowser) {` ... `                var targetEl = targetNode.firstChild;
                while(targetEl) {
                    var id = targetEl.id;`
yes, i know it's a terrible dom structure (working on it, eventually ;) ), but for some reason this seems like the only component which has this problem, in many other place i'm using the "dangling style tag" structure w/o issue.
moving the <style> into the <div> seems to make it not get confused ... i could try to get a smaller repro and file an issue if necessary but wanted to share as i found it.  

i didn't pursue it too much b/c i had a workaround (dump `<style>` inside a single top-level `<div>` tag per component instead of outside) and i'm planning in revamping all my styles eventually anyway

**Comment by smulder** [comment]
I just came across this same bug to add a second to the vote... moving the style tag into the root div of the template fixed it for now.

**Comment by austinkelleher** [comment]
Fixed in `marko@4.4.18`
